Include/internal/pycore_brc.h

```diff
@@ -0,0 +1,74 @@
+#ifndef Py_INTERNAL_BRC_H
+#define Py_INTERNAL_BRC_H
+
+#include <stdint.h>
+#include "pycore_llist.h"           // struct llist_node
+#include "pycore_lock.h"            // PyMutex
+#include "pycore_object_stack.h"    // _PyObjectStack
```

Comment:
Why not put these after the `#ifndef Py_BUILD_CORE` block?

Reply:
It seems like we've been okay in other header files.  Did something change?  Perhaps we simply weren't aware of the problem?  Should we go through our other header files and change to this pattern?

Reply:
The problem only exists if these header files are compiled as C++ code. It's a bit weird because these are *internal* header files and CPython is in C (not C++), but we've added `extern "C" {` to them so it seems like we intend to support C++ compilation. Sometimes embedders or naughty C-API extensions include the `pycore_xxx` headers (possibly in a C++ file).

I think we should fix the other internal headers, but no one has complained yet. Most likely that's because:

1) It's rare to compile one of these headers as C++
2) Most of the `pycore_xxx` headers don't include system headers.

Here's an example of the same issue in our public header files, which Victor fixed: https://github.com/python/cpython/issues/110845#issuecomment-1805975034

Reply:
ping @gpshead

I'm sure you have some insight on this. 😄

Reply:
I filed https://github.com/python/cpython/issues/115105 to track and discuss this